src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCIService.java

```diff
@@ -38,20 +38,22 @@ public class LocalCIService extends AbstractContinuousIntegrationService {
 
     private final LocalCIDockerService localCIDockerService;
 
-    @Value("${artemis.continuous-integration.build.images.java.default}")
-    String dockerImage;
+    private final ProgrammingLanguageConfiguration programmingLanguageConfiguration;
 
     public LocalCIService(ProgrammingSubmissionRepository programmingSubmissionRepository, FeedbackRepository feedbackRepository, BuildLogEntryService buildLogService,
-            BuildLogStatisticsEntryRepository buildLogStatisticsEntryRepository, TestwiseCoverageService testwiseCoverageService, LocalCIDockerService localCIDockerService) {
+            BuildLogStatisticsEntryRepository buildLogStatisticsEntryRepository, TestwiseCoverageService testwiseCoverageService, LocalCIDockerService localCIDockerService,
+            ProgrammingLanguageConfiguration programmingLanguageConfiguration) {
         super(programmingSubmissionRepository, feedbackRepository, buildLogService, buildLogStatisticsEntryRepository, testwiseCoverageService);
         this.localCIDockerService = localCIDockerService;
+        this.programmingLanguageConfiguration = programmingLanguageConfiguration;
     }
 
     @Override
     public void createBuildPlanForExercise(ProgrammingExercise programmingExercise, String planKey, VcsRepositoryUrl sourceCodeRepositoryURL, VcsRepositoryUrl testRepositoryURL,
             VcsRepositoryUrl solutionRepositoryURL) {
         // Only check whether the docker image needed for the build plan exists.
-        localCIDockerService.pullDockerImage(dockerImage);
+        localCIDockerService.pullDockerImage(
+                programmingLanguageConfiguration.getImage(programmingExercise.getProgrammingLanguage(), Optional.ofNullable(programmingExercise.getProjectType())));
```

Comment:
```suggestion
        localCIDockerService.pullDockerImage(
            programmingLanguageConfiguration.getImage(
                programmingExercise.getProgrammingLanguage(), 
                Optional.ofNullable(programmingExercise.getProjectType())
            )
        );
```
(Not sure if I got the indentation exactly right, but) please put split up line 56 into multiple lines, as it is currently very long.